Functions/Administration/xASL_adm_SortDicomToFolders.m

```diff
@@ -57,50 +57,51 @@ function xASL_adm_SortDicomToFolders(pathDICOM, nDirLayers, bUseDCMTK, bVerbose)
 			% Read the DICOM file - all error handling is inside this function
 			tDcm = xASL_io_DcmtkRead(Flist{iL}, false, bUseDCMTK, true);
 
+			hasProtocolName = isfield(tDcm, 'ProtocolName') && ~isempty(tDcm.ProtocolName);
+			hasSeriesDescription = isfield(tDcm, 'SeriesDescription') && ~isempty(tDcm.SeriesDescription);
+			hasSeriesNumber = isfield(tDcm, 'SeriesNumber') && ~isempty(tDcm.SeriesNumber);
+
 			% Check that the header was read and contains the basic tags
 			if isempty(tDcm)
                 warning(['Empty DICOM header, skipping: ' Flist{iL}]);
-                hasProtocolName = isfield(tDcm, 'ProtocolName') && ~isempty(tDcm.ProtocolName);
-                hasSeriesDescription = isfield(tDcm, 'SeriesDescription') && ~isempty(tDcm.SeriesDescription);
-                hasSeriesNumber = isfield(tDcm, 'SeriesNumber') && ~isempty(tDcm.SeriesNumber);
-            elseif ~hasProtocolName || ~hasSeriesDescription || ~hasSeriesNumber
+			elseif ~hasProtocolName || ~hasSeriesDescription || ~hasSeriesNumber
                 warning(['DICOM header without ProtocolName, SeriesDescription, or SeriesNumber, skipping: ' Flist{iL}]);
-
+			else
                 % Manage directory name
                 % Priority:
                 % 1. ProtocolName
                 % 2. SeriesDescription
                 % 3. SeriesNumber
 
                 Fname = [];
-                foundName = false;
 
                 % Always add the protocol name to the directory name
                 if hasProtocolName
                     Fname = tDcm.ProtocolName;
-                    foundName = true;
                 end
                 
 				% Add series description if available
 				if hasSeriesDescription
-                    if foundName && ~strcmpi(tDcm.ProtocolName, tDcm.SeriesDescription)
+                    if ~isempty(Fname) && ~strcmpi(tDcm.ProtocolName, tDcm.SeriesDescription)
```

Comment:
Should be if hasProtocolName instead. I know it now is the same, but if anyone edits the function, hasProtocolName is clearer

Reply:
Done